ts/components/conversation/composition/CompositionBox.tsx

```diff
@@ -424,16 +424,27 @@ class CompositionBoxInner extends React.Component<Props, State> {
       return null;
     }
 
-    const { isKickedFromGroup, left, isPrivate, isBlocked } = this.props.selectedConversation;
-    const messagePlaceHolder = isKickedFromGroup
-      ? i18n('youGotKickedFromGroup')
-      : left
-      ? i18n('youLeftTheGroup')
-      : isBlocked && isPrivate
-      ? i18n('unblockToSend')
-      : isBlocked && !isPrivate
-      ? i18n('unblockGroupToSend')
-      : i18n('sendMessage');
+    const {
+      isKickedFromGroup,
+      left,
+      isPrivate,
+      isBlocked,
+      didApproveMe,
+      isApproved,
+    } = this.props.selectedConversation;
+    const messagePlaceHolder =
+      // isApproved && !didApproveMe && isPrivate
+      isApproved && !didApproveMe && isPrivate
+        ? i18n('messageRequestPending')
+        : isKickedFromGroup
+        ? i18n('youGotKickedFromGroup')
+        : left
+        ? i18n('youLeftTheGroup')
+        : isBlocked && isPrivate
+        ? i18n('unblockToSend')
+        : isBlocked && !isPrivate
+        ? i18n('unblockGroupToSend')
+        : i18n('sendMessage');
```

Comment:
can you move this to a function with some real ifs and some comments between each steps so later update are easier/